Tensile/KernelWriterSource.py

```diff
@@ -3311,7 +3311,11 @@ def wait(self, kernel, tPA, tPB, globalRead, localWrite, localRead, comment):
   # SyncThreads
   ##############################################################################
   def syncThreads(self, kernel, comment=""):
-    return self.indent + self.syncStr + " //" + comment + self.endLine
+    if kernel["NumThreads"] > kernel["WavefrontSize"]:
+        return self.indent + self.syncStr + " //" + comment + self.endLine
+    else:
+        return self.indent + "// Skip barrier: NumThreads=%s"%(kernel["NumThreads"]) + \
+              self.indent + self.endLine
```

Comment:
Do we need self.indent again here?

Reply:
Yes. Otherwise, the comment line has different indent. @nakajee

Reply:
I know, but second self.indent seems not necessary before self.endLine.
We do not need any indent for endLine, right?

Reply:
@nakajee right. I forgot removing it. Done.